testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1347,9 +1347,20 @@ public GradleExecuter withBuildCacheEnabled() {
     }
 
     protected Action<ExecutionResult> getResultAssertion() {
+        boolean shouldCheckDeprecations = checkDeprecations;
+
+        // Rich consoles mess with our deprecation log scraping
+        boolean isAuto = consoleType == null || consoleType == ConsoleOutput.Auto;
+        if ((isAuto && consoleAttachment != ConsoleAttachment.NOT_ATTACHED) ||
+            consoleType == ConsoleOutput.Verbose ||
+            consoleType == ConsoleOutput.Rich
+        ) {
+            shouldCheckDeprecations = false;
```

Comment:
💭 It would be nice if your test executor immediately failed if you tried to run with rich console and told it to expect any deprecations.